ProteoformSuiteInternal/Lollipop.cs

```diff
@@ -829,42 +828,42 @@ public static void calibrate_files()
             filenames.AddRange(input_files.Where(f => f.purpose == Purpose.CalibrationIdentification).Select(f => f.filename).Distinct());
             filenames.AddRange(Lollipop.td_hits_calibration.Select(f => f.filename).Distinct());
             //get calibration points and calibrate deconvolution files
-            Parallel.ForEach(filenames.Distinct(), filename =>
-                {
-                    get_calibration_points(filename);
-                    calibrate_td_hits(filename);
-                    determine_component_shift(input_files.Where(f => f.purpose == Purpose.CalibrationIdentification && f.filename == filename).ToList());
-                    InputFile file = input_files.Where(f => f.purpose == Purpose.CalibrationIdentification && f.filename == filename).FirstOrDefault();
-                    if (file != null) Calibration.calibrate_components_in_xlsx(file);
-                });
+            foreach (string filename in filenames.Distinct())
+            {
+                get_calibration_points(filename);
+                calibrate_td_hits(filename);
+                determine_component_shift(input_files.Where(f => f.purpose == Purpose.CalibrationIdentification && f.filename == filename).ToList());
+                InputFile file = input_files.Where(f => f.purpose == Purpose.CalibrationIdentification && f.filename == filename).FirstOrDefault();
+                if (file != null) Calibration.calibrate_components_in_xlsx(file);
+            }
             foreach (InputFile file in Lollipop.input_files.Where(f => f.purpose == Purpose.CalibrationTopDown))
             {
                 Calibration.calibrate_td_hits_file(file);
             }
         }
-    
 
-    public static void get_calibration_points(string filename)
+
+        public static void get_calibration_points(string filename)
         {
             List<InputFile> raw_files = Lollipop.input_files.Where(f => f.purpose == Purpose.RawFile && f.filename == filename).ToList();
-            if (raw_files.Count > 0)
+            if (raw_files.Count == 1)
```

Comment:
What if two different files had the same filenames in two different folders? It's a weird case, but it seems like you should check the complete path instead of the filename here.